examples/vault-cluster-private/main.tf

```diff
@@ -68,6 +68,23 @@ data "template_file" "user_data_vault_cluster" {
   }
 }
 
+# ---------------------------------------------------------------------------------------------------------------------
+# PERMIT CONSUL SPECIFIC TRAFFIC IN VAULT CLUSTER
+# To allow our Vault servers consul agents to communication with consul agents and servers, we open up the consul
+# specific protocols and ports for consul traffic
+# ---------------------------------------------------------------------------------------------------------------------
+
+module "security_group_rules" {
+  source = "github.com/hashicorp/terraform-aws-consul.git//modules/consul-security-group-rules?ref=v0.3.1"
```

Comment:
Hm, I'm worried this might be overkill, and a bit of a security risk. When Consul is running in client mode, as it is on the Vault nodes, it should only listen on localhost to most ports. The only port that should be exposed to the outside world is the LAN gossip port. See [Consul Architecture](https://www.consul.io/docs/internals/architecture.html) for details.

That makes me think we should have a new, very simple module in `terraform-aws-consul` called `consul-client-security-group-rules` that opens up just this one port...

Reply:
You're absolutely right, I just took the simplest route however opening up all that traffic _on the vault instances_ is not a good idea! I will modify this and submit that small module change on the `terraform-aws-consul` repo.